test/index.test.js

```diff
@@ -156,4 +156,52 @@ describe('index', () => {
                     `${templateConfig.name}@${templateConfig.version} was successfully published`));
         });
     });
+
+    describe('Template Tag', () => {
+        const config = {
+            name: 'template/test',
+            tag: 'stable',
+            version: '1.0.0'
+        };
+
+        it('throws error when request yields an error', () => {
+            requestMock.rejects(new Error('error'));
+
+            return index.tagTemplate(config)
+                .then(() => assert.fail('should not get here'))
+                .catch(err => assert.equal(err.message, 'error'));
+        });
+
+        it('throws error for the corresponding request error status code if not 201', () => {
+            const responseFake = {
+                statusCode: 403,
+                body: {
+                    statusCode: 403,
+                    error: 'Forbidden',
+                    message: 'Fake forbidden message'
+                }
+            };
+
+            requestMock.resolves(responseFake);
+
+            return index.tagTemplate(config)
+                .then(() => assert.fail('should not get here'))
+                .catch(err => assert.equal(err.message,
+                    'Error tagging template. 403 (Forbidden): Fake forbidden message')
+                );
+        });
+
+        it('succeeds and does not throw an error if request status code is 201', () => {
+            const responseFake = {
+                statusCode: 201,
+                body: templateConfig
+            };
+
+            requestMock.resolves(responseFake);
+
+            return index.tagTemplate(config)
```

Comment:
Why don't you assert that the `requestMock` was called correctly? Since templates do not namespace, if you have a template name of `tkyi/myTemplate`, it should be calling the endpoint `templates/tkyi%2FmyTemplate/{tagName}` and not `templates/tkyi/myTemplate/{tagName}`